### ISSUE TITLE
New rust plugin fails to validate with a `rust-deps` part dependency

### ISSUE THREAD

**Issue by lengau** [canonical/craft-parts#559]
### Bug Description

We allow a `rust-deps` part dependency in the rust plugin so the user may install rust in the manner they choose.

The updated rust plugin, however, fails to validate when this occurs.

### To Reproduce

Modify the `test_get_pull_commands_compat` test in `test_rust_plugin.py` as follows:

```python
def test_get_pull_commands_compat(part_info):
    properties = RustPlugin.properties_class.unmarshal(
        {"source": ".", "after": ["rust-deps"]}
    )
    plugin = RustPlugin(properties=properties, part_info=part_info)
    plugin.validator_class(
        part_name="project",
        env="",
        properties=properties
    ).validate_environment(part_dependencies=["rust-deps"])
    plugin._check_rustup = lambda: False

    commands = plugin.get_build_commands()
    assert plugin.get_pull_commands() == []
    assert 'cargo install -f --locked --path "."' in commands[0]
```

### part yaml

```shell
parts:
  rust-deps:
    plugin: nil
  project:
    source: .
    after: ["rust-deps"]
```

### Relevant log output

```shell
FAILED   [100%]
tests/unit/plugins/test_rust_plugin.py:184 (test_get_pull_commands_compat)
part_info = <craft_parts.infos.PartInfo object at 0x7fdbcf9c8850>

    def test_get_pull_commands_compat(part_info):
        properties = RustPlugin.properties_class.unmarshal(
            {"source": ".", "after": ["rust-deps"]}
        )
        plugin = RustPlugin(properties=properties, part_info=part_info)
>       plugin.validator_class(
            part_name="project",
            env="",
            properties=properties
        ).validate_environment(part_dependencies=["rust-deps"])

/home/lengau/Work/Code/craft-parts/tests/unit/plugins/test_rust_plugin.py:190: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
/home/lengau/Work/Code/craft-parts/craft_parts/plugins/rust_plugin.py:103: in validate_environment
    options.rust_channel = "none"  # type: ignore[misc]
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

>   ???
E   TypeError: "RustPluginProperties" is immutable and does not support item assignment

pydantic/main.py:359: TypeError
```

### Other information
When fixing this bug, remove the type  linter ignore comment in `rust_plugin.py` that mentions this bug.

**Comment by cmatsuoka** [comment]
@liushuyu Could you check this issue please? This is a good scenario to test as well.

**Comment by liushuyu** [comment]
Fix in progress at #560
